### PR TITLE
megahit_toolkit contig2fastg

### PR DESCRIPTION
**Pull request by jraysajulga**
> This tool wrapper was originally uploaded to the [galaxyproteomics/tools-galaxyp repository](https://github.com/galaxyproteomics/tools-galaxyp/tree/master/tools/megahit_contig2fastg) in this [pull request](https://github.com/galaxyproteomics/tools-galaxyp/pull/322), but I figured it would be better to have this tool alongside IUC's MEGAHIT.

**Contig2fastg is a subprogram of MEGAHIT toolkit.** It converts contigs (.fa) from MEGAHIT into assembly graphs (.fastg). The GalaxyP team is looking to use this tool within a metatranscriptomics workflow to generate assembly graphs for use in [Graph2Pro](https://github.com/COL-IU/graph2pro-var).

The .fastg datatype is currently being implemented in this [pull request](https://github.com/galaxyproject/galaxy/pull/6987).

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

FOR REVIEWER:
* [x] .shed.yml file ok
    - [x] Toolshed user `iuc` has access to associated toolshed repo(s)
* [x] Indentation is correct (4 spaces)
* [x] Tool version/build ok
* [x] `<command/>`
  - [x] Text parameters, input and output files `'single quoted'`
  - [x] Use of `<![CDATA[ ... ]]>` tags
  - [ ] Parameters of type `text` or having `optional="true"` attribute are checked with `if str($param)` before being used
* [x] Data parameters have a `format` attribute containing datatypes recognised by Galaxy
* [x] Tests
  - [x] Parameters are reasonably covered
  - [ ] Test files are appropriate
* [x] Help
  - [ ] Valid restructuredText and uses `<![CDATA[ ... ]]>` tags
* [ ] Complies with other best practice in [Best Practices Doc](http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html)

### CODE REVIEW

File: tools/megahit_contig2fastg/megahit_contig2fastg.xml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<?xml version='0.9' encoding='utf-8'?>
```
Comment (Slugger70) [inline tools/megahit_contig2fastg/megahit_contig2fastg.xml]: Would you consider putting the megahit_toolkit version here and the galaxy wrapper version? 

It is convenient to tokenize the tool version and then add +galaxy0; the latter can be increased when only the wrapper is updated but not underlying tool. And when the underlying tool version is updated, only the macros file would need to be changed:

Comment (jraysajulga) [reply tools/megahit_contig2fastg/megahit_contig2fastg.xml]: Sure. I'm not entirely sure where to put the versions though. I've placed the megahit_toolkit version within the "macros" tag; do I put the galaxy wrapper version in the tool tag?

I've also thought of removing the first line:

`<?xml version='1.0' encoding='utf-8'?>`

Comment (jraysajulga) [reply tools/megahit_contig2fastg/megahit_contig2fastg.xml]: Okay. Thanks for the help. I just implemented those changes in the commits below.

**Comment by jraysajulga** [comment]
Is this good for merging? Are there any further changes that I need to make?

**Comment by bgruening** [comment]
@Slugger70 approved it so yes, this can be merged.

**Pull request merged**